File: projects/06/06_python/test_code.py
import unittest
import parser

class TestCode(unittest.TestCase):
    def setUp(self):
       self.test_code = parser.Code()
    
    def test_dest(self):
        test = self.test_code.dest("null")
        self.assertEqual(test, '000')

    def test_comp(self):
        test = self.test_code.comp("0")
        self.assertEqual(test, '0101010')

    def test_jump(self):
        test = self.test_code.jump("null")
        self.assertEqual(test, '000')
        
if __name__ == "__main__":
    unittest.main()

File: projects/06/Assembler/src/TestCode.java
import junit.framework.*;
import org.junit.Test;

// This isn't used but keeping around just in case (for now)
//import junit.extensions.TestSetup; 


public class TestCode extends TestCase {
	
	Code myCode = new Code();

	@Test
	public void testComp() {
	
		assertEquals("0101010", myCode.comp("0"));
	}
	
	@Test
	public void testDest() {

		assertEquals("001", myCode.dest("M"));
	}

	@Test
	public void testJump() {
		// "M","001"
		assertEquals("001", myCode.jump("JGT"));
	}


}
